lib/solargraph/rbs_map/conversions.rb

```diff
@@ -29,11 +29,37 @@ def type_aliases
         @type_aliases ||= {}
       end
 
+      # @param loader [RBS::EnvironmentLoader]
+      # @return [void]
+      def load_environment_to_pins(loader)
+        environment = RBS::Environment.from_loader(loader).resolve_type_names
+        cursor = pins.length
+        environment.declarations.each { |decl| convert_decl_to_pin(decl, Solargraph::Pin::ROOT_PIN) }
+        added_pins = pins[cursor..-1]
+        add_back_implicit_pins(added_pins)
+      end
+
+      # @param added_pins [Range<Pin>]
+      # @return [void]
+      def add_back_implicit_pins(added_pins)
+        added_pins.each do |pin|
+          pin.source = :rbs
+          next unless pin.is_a?(Pin::Namespace) && pin.type == :class
+          next if pins.any? { |p| p.path == "#{pin.path}.new"}
+          pins.push Solargraph::Pin::Method.new(
+                      location: nil,
+                      closure: pin,
```

Comment:
This line changed from closure.pin to pin.  We are being passed the pin for the class; we don't want to create the initializer in the class' surrounding closure, but in the class we were passed directly, as a class method.

Reply:
I would guess at some point we'd want to teach Solargraph to chase the parent classes to look for the first one with an initializer and use its arity and types, replacing its return type, rather than assuming the only one defined was Object.new.  

Let me know if I'm misunderstanding anything.

Reply:
That would definitely be more helpful. The implementation might be tricky. I'm guessing we would either have to update the initializers after all the pins have been generated or figure out a way to resolve them lazily.